pkgs/tools/misc/colorless/default.nix

```diff
@@ -0,0 +1,47 @@
+{ lib
+, fetchurl
+, stdenvNoCC
+, coreutils
+, dash
+, gnused
+, less
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "colorless";
+  version = "109";
+
+  src = fetchurl {
+    url = "http://software.kimmo.suominen.com/${pname}-${version}.tar.gz";
+    sha256 = "039a140fa11cf153cc4d03e4f753b7ff142cab88ff116b7600ccf9edee81927c";
+  };
+
+  makeFlags = [
+    "TOOLPATH=${coreutils}/bin:${gnused}/bin:${less}/bin"
+    "PREFIX=${placeholder "out"}"
+    "SHELL=${dash}/bin/dash"
+  ];
+
+  preInstall = ''
+    mkdir -p $out/bin $out/share/man/man1
+  '';
```

Comment:
Shouldn't this be fixed in the package's Makefile instead? `make install` should really be making any directories it needs if they're not present.

Reply:
With modern packaging systems that's probably the best practice. I can make that change in the next version of `colorless` and across other tools I maintain.

On legacy systems `/usr/local/bin` and `/usr/local/man/man1` had many different practices in the past: most notably the question of having writability by the `staff` group or not. I felt that it was better to error out during `make install` than create them with the "wrong" permissions.